source/agora/test/Base.d

```diff
@@ -492,6 +498,34 @@ public interface TestAPI : ValidatorAPI
     public abstract void broadcastPreimage (uint height);
 }
 
+/*******************************************************************************
+
+    For dependency injection.
+
+*******************************************************************************/
+
+public class PreLoadedMemBlockStorage : MemBlockStorage
+{
+    /// Preloaded blocks, including the genesis block
+    immutable(Block)[] blocks;
+
+    /// Ctor
+    public this (immutable(Block)[] blocks)
+    {
+        this.blocks = blocks;
+    }
+
+    /// Load the blockchain
+    public override bool load ()
+    {
+        foreach (const ref block; this.blocks)
+            this.saveBlock(block);
+
+        return true;
+    }
+}
```

Comment:
Does it makes sense to have it here ? That could just replace the MemBlockStorage, couldn't it ? Do you remember why `MemBlockStorage` has serialized Blocks ?

Reply:
> Does it makes sense to have it here ? 

Ah it should probably not be `public`. I can move it inside of `TestNodeMixin` I suppose. I only have one use-case for it and it's here in the tests.

> That could just replace the MemBlockStorage, couldn't it ? Do you remember why MemBlockStorage has serialized Blocks ?

Not sure. I think we can just remove that part? Then I could just add a constructor to `MemBlockStorage` and get rid of `PreLoadedMemBlockStorage` I suppose.